app/assets/javascripts/libs/utils.js

```diff
@@ -244,6 +244,51 @@ const Utils = {
   toNullable<T>(maybe: Maybe<T>): ?T {
     return maybe.isJust ? maybe.get() : null;
   },
+
+  getRecursiveKeysAndValues(obj: Object): Array<any> {
+    return _.flattenDeep(Utils.getRecursiveKeysAndValuesUnflat(obj));
+  },
+
+  getRecursiveKeysAndValuesUnflat(obj: Object): Array<any> {
+    if (_.isArray(obj)) {
+      return obj.map(Utils.getRecursiveKeysAndValuesUnflat);
+    } else if (_.isObject(obj)) {
+      return Object.keys(obj).map(key => [key, Utils.getRecursiveKeysAndValuesUnflat(obj[key])]);
+    } else {
+      return [obj];
+    }
+  },
+
+  // Filters an array given a search string. Supports regex seach and several words as OR query.
```

Comment:
I think this is wrong since the code escapes special characters like [], () etc.